Sources/FluidAudio/Diarizer/SegmentationProcessor.swift

```diff
@@ -1,55 +1,177 @@
+import Accelerate
 import CoreML
 import Foundation
 import OSLog
 
+/// Segmentation processor with ANE-aligned memory and zero-copy operations
 @available(macOS 13.0, iOS 16.0, *)
 public struct SegmentationProcessor {
 
     private let logger = Logger(subsystem: "com.fluidinfluence.diarizer", category: "Segmentation")
+    private let memoryOptimizer = ANEMemoryOptimizer.shared
 
     public init() {}
 
     func getSegments(
-        audioChunk: ArraySlice<Float>, segmentationModel: MLModel, chunkSize: Int = 160_000
-    ) throws -> [[[Float]]] {
+        audioChunk: ArraySlice<Float>,
```

Comment:
Might be best to add some unit tests for these changes too, including the clustering changes

Reply:
Try to capture edge cases in regards to memory usage, they're quite fragile to changes